### PR TITLE
React components

### PR DESCRIPTION
**Pull request by Doooodie**
1. Task: https://github.com/rolling-scopes-school/tasks/tree/master/react/modules/module01
2. Screenshot:
image
3. Done 20.03.2023 / deadline 20.03.2023
4. Score: 15 / 15

    - Linting can be run and doesn't produce any errors or warnings. `+3`
    - Tests can be run and the test coverage is equal or higher than 40%. `+3`
    - Routing is enabled, header shows current route. There are 404 and 'About us' routes. `+4`
    - Search bar is on the main page, value from the search is saved in LocalStorage on component unmount. `+4`
    - Cards are present on the main page. `+1`

### CODE REVIEW

File: src/main.tsx [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import { createRoot } from 'react-dom/client';
+
+import App from './components/App';
+
+createRoot(document.getElementById('root') as HTMLElement).render(<App />);
```
Comment (saha80) [inline src/main.tsx:5]: Your app should be using React.StrictMode.
StrictMode lets you find common bugs in your components early during development.
legacy docs, reference.

File: src/components/pages/Home/Cards/Cards.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  products: IProduct[];
+};
+
+class Cards extends Component<unknown, State> {
+  state: State = {
+    products: [],
+  };
+
+  async componentDidMount() {
+    const query = await fetch('https://dummyjson.com/products');
```
Comment (saha80) [inline src/components/pages/Home/Cards/Cards.tsx:17]: You should move the fetching process to a "CardsService.ts" in the "services" folder.

https://blog.webdevsimplified.com/2022-07/react-folder-structure/

[the remark this answers is not in the document]
Comment (Doooodie) [reply src/components/pages/Home/Cards/Cards.tsx:17]: There's no requirements in task about using fetch and especially handling error boundaries. Ofc I'll add them if needed.

**Comment by saha80** [comment]
The design of the application is nice and userfriendly.

**Pull request closed without merging**